src/odemis/acq/fastem.py

```diff
@@ -223,7 +223,13 @@ def acquire(roa, path, scanner, multibeam, descanner, detector, stage, ccd, beam
                 (model.DataArray): The acquisition data, which depends on the value of the detector.dataContent VA.
                 (Exception or None): Exception raised during the acquisition or None.
     """
-    f = model.ProgressiveFuture()
+
+    est_dur = roa.estimate_acquisition_time()
+    if pre_calibrate:
+        est_dur += 60  # add some extra time for the pre-calibrations
+        # TODO replace with call to estimate calibration time in module for each calibrations
+        #  when code here replaced with calls to calibration manager
+    f = model.ProgressiveFuture(start=time.time(), end=time.time() + est_dur)
```

Comment:
The time estimate is currently overridden by line 1621 in gui/cont/acquisition.py. I don't have a good idea how to handle this yet. The ROA, which we get the time estimate from, does not know about whether pre calibrations need to be run or not.

Reply:
You could add `pre_calibrate` as an attribute to the ROA class, and then update `roa.estimate_acquisition_time()` to add extra time when `pre_calibrate` is True. Then you could remove it as argument from `acquire` and `AcquisitionTask`, line 345 would then become `if self._roa.pre_calibrate:`

Reply:
you will then also need to update `test_pre_calibrate`

Reply:
Yes that would make sense. I'll wait for the review from Éric whether I now have put setting the time in the right places and then I will change this.

Reply:
That sounds like a good plan to me too.